Services/Simulations.cs

```diff
@@ -698,5 +701,31 @@ private async Task<Models.Simulation> ResetSimulationStatisticsAsync(Models.Simu
 
             return simulation;
         }
+
+        private async Task ResetActiveDevicesStatistics(Models.Simulation simulation)
+        {
+            try
+            {
+                var currentStats = await this.simulationStatistics.GetSimulationStatisticsAsync(simulation.Id);
+
+                if (currentStats != null)
+                {
+                    var simulationModel = new SimulationStatisticsModel
+                    {
+                        ActiveDevices = 0, // Reset active devices to 0
+                        TotalMessagesSent = currentStats.TotalMessagesSent,
+                        FailedMessages = currentStats.FailedMessages,
+                        FailedDeviceConnections = currentStats.FailedDeviceConnections,
+                        FailedDevicePropertiesUpdates = currentStats.FailedDevicePropertiesUpdates,
+                    };
+
+                    await this.simulationStatistics.UpdateAsync(simulation.Id, simulationModel);
+                }
+            }
+            catch (Exception e)
+            {
+                this.log.Error("Error updating active device count statistics.", () => new { simulation.Id, e });
```

Comment:
do we need to log this into diagnostics logger as well?

Reply:
same.. we dont need to log this.